stdlib/grpc/src/main/java/org/ballerinalang/net/grpc/builder/BallerinaFileBuilder.java

```diff
@@ -174,32 +198,43 @@ private void computeSourceContent(byte[] descriptor, String mode) throws CodeBui
-                            SAMPLE_CLIENT_TEMPLATE_NAME, clientFilePath);
-                }
+
                 if (GRPC_SERVICE.equals(mode)) {
+                    serviceFile = sampleServiceBuilder.build();
+                    if (!needStubFile) {
+                        serviceFile.setMessageList(messageList);
+                        serviceFile.setEnumList(enumList);
+                        serviceFile.setDescriptors(descriptors);
+                        if (!stubRootDescriptor.isEmpty()) {
+                            serviceFile.setRootDescriptor(stubRootDescriptor);
+                        }
+                    }
                     String servicePath = generateOutputFile(this.balOutPath, serviceDescriptor.getName() +
                             SAMPLE_SERVICE_FILE_PREFIX);
-                    writeOutputFile(sampleServiceBuilder.build(), DEFAULT_SAMPLE_DIR, SAMPLE_SERVICE_TEMPLATE_NAME,
+                    writeOutputFile(serviceFile, DEFAULT_SAMPLE_DIR, SAMPLE_SERVICE_TEMPLATE_NAME,
                             servicePath);
+                } else if (GRPC_CLIENT.equals(mode)) {
+                    String clientFilePath = generateOutputFile(
+                            this.balOutPath,
+                            serviceDescriptor.getName() + SAMPLE_FILE_PREFIX
+                    );
+                    writeOutputFile(new ClientFile(serviceDescriptor.getName(), isUnaryContains),
+                            DEFAULT_SAMPLE_DIR,
+                            SAMPLE_CLIENT_TEMPLATE_NAME, clientFilePath);
+
                 }
             }
-            // read message types.
-            for (DescriptorProtos.DescriptorProto descriptorProto : messageTypeList) {
-                Message message = Message.newBuilder(descriptorProto).build();
-                stubFileObject.addMessage(message);
-            }
-            // read enum types.
-            for (DescriptorProtos.EnumDescriptorProto descriptorProto : enumDescriptorProtos) {
-                EnumMessage enumMessage = EnumMessage.newBuilder(descriptorProto).build();
-                stubFileObject.addEnumMessage(enumMessage);
-            }
 
-            String stubFilePath = generateOutputFile(this.balOutPath, filename + STUB_FILE_PREFIX);
-            writeOutputFile(stubFileObject, DEFAULT_SKELETON_DIR, SKELETON_TEMPLATE_NAME, stubFilePath);
+            if (!GRPC_SERVICE.equals(mode) || needStubFile) {
+                stubFileObject.setMessageList(messageList);
+                stubFileObject.setEnumList(enumList);
+                stubFileObject.setDescriptors(descriptors);
+                if (!stubRootDescriptor.isEmpty()) {
+                    stubFileObject.setRootDescriptor(stubRootDescriptor);
+                }
+                String stubFilePath = generateOutputFile(this.balOutPath, filename + STUB_FILE_PREFIX);
+                writeOutputFile(stubFileObject, DEFAULT_SKELETON_DIR, SKELETON_TEMPLATE_NAME, stubFilePath);
```

Comment:
Shall we create a new service stub template file with the only message types and descriptors?
and generate that stub file when it is service mode. Because we don't need stub functions in service mode.

Reply:
Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/21024/commits/f18fc70da127f866e12b0df7937719b6a4259a1b